stake-pool/program/src/processor.rs

```diff
@@ -845,7 +870,30 @@ impl Processor {
             return Err(ProgramError::IncorrectProgramId);
         }
 
-        let mut stake_pool = State::deserialize(&stake_pool_info.data.borrow())?.stake_pool()?;
+        let mut stake_pool = StakePool::deserialize(&stake_pool_info.data.borrow())?;
+        if !stake_pool.is_initialized() {
+            return Err(StakePoolError::InvalidState.into());
+        }
+
+        // TODO: remove conditional compilation when time travel in tests is possible
+        #[cfg(not(feature = "test-bpf"))]
+        {
+            let stake_acc_state: stake::StakeState =
+                deserialize(&stake_info.data.borrow()).unwrap();
+            let delegation = stake_acc_state.delegation();
+            if let Some(delegation) = delegation {
+                let target_epoch = clock.epoch;
+                let history = None;
+                let fix_stake_deactivate = false;
+                let (effective, activating, deactivating) = delegation
+                    .stake_activating_and_deactivating(target_epoch, history, fix_stake_deactivate);
+                if activating != 0 && deactivating != 0 && effective == 0 {
```

Comment:
This should probably be 
```suggestion
                if activating != 0 || deactivating != 0 || effective == 0 {
```

Reply:
Right, this is what inability to test results in, good catch.